cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureMetadataCollector.java

```diff
@@ -129,9 +130,51 @@ private String getLocalityIndicator(Integer faultDomainCount, CloudContext cloud
     }
 
     @Override
-    public List<CloudLoadBalancerMetadata> collectLoadBalancer(AuthenticatedContext ac, List<LoadBalancerType> loadBalancerTypes) {
-        // no-op
-        return Collections.emptyList();
-    }
+    public List<CloudLoadBalancerMetadata> collectLoadBalancer(AuthenticatedContext ac, List<LoadBalancerType> loadBalancerTypes,
+            List<CloudResource> resources) {
+        LOGGER.debug("Collecting Azure load balancer metadata, for cluster {}", ac.getCloudContext().getName());
+
+        List<CloudLoadBalancerMetadata> cloudLoadBalancerMetadata = new ArrayList<>();
+        String resourceGroup = azureUtils.getTemplateResource(resources).getName();
+        AzureClient azureClient = ac.getParameter(AzureClient.class);
+
+        for (LoadBalancerType type : loadBalancerTypes) {
+            String loadBalancerName = AzureLoadBalancer.getLoadBalancerName(type);
+            LOGGER.debug("Attempting to collect metadata for load balancer {}, type {}", loadBalancerName, type);
+            try {
+                String ip = null;
+                if (LoadBalancerType.PUBLIC.equals(type)) {
+                    List<String> publicIps = azureClient.getLoadBalancerIps(resourceGroup, loadBalancerName);
+                    if (publicIps.isEmpty()) {
+                        LOGGER.warn("Unable to find public ip address for load balancer {}", loadBalancerName);
+                    } else {
+                        ip = publicIps.get(0);
```

Comment:
It's my understanding that there _can_ be multiple public IP addresses, I'm not sure how to handle it exactly though.

We're plugging the IP address here into the `CloudLoadBalancerMetadata` which is then used to configure DNS records.

I think the risk here would be that we add multiple IP addresses, use the first one with registration, and then our bookkeeping gets messed up and the public IP is removed.

We could consider keeping _all_ of the IP addresses in the metadata, but that feels like a lot of code when I'm not sure exactly what the use case is.

@hreeve-cloudera do you have any other thoughts about where picking the first public IP might bite us?

Reply:
And, do I have it right about the usage of the IP -- *we're just using it for DNS registration*?

Reply:
Thanks, if that is the case, we might consider validating if there is _exactly_ 1 public ip as expected, as an extra clause in line 149, just to be sure.

Reply:
I've added a `warn` log if `publicIps.size() > 1`.

If more than 1 IP _is_ available, we'll have to come back here and add more sophisticated selection logic. But, I think it would be premature to do that now.